### DISCUSSION TITLE
NextJS dynamic catch all w/wildcard pages in SXA starter for XM cloud

### DISCUSSION THREAD

**Discussion by Thomas-Ballard** [Sitecore/jss#1490] [Help] [answered]
### Description

Hello,

I wonder if anyone has run into this issue previously. I am kicking off a project using the https://github.com/sitecorelabs/xmcloud-foundation-head project and wanted to create a dynamic subroute to contain events pages, which will be generated from generic content from a wildcard page in XM cloud, enriched by a query about the specific event from a 3rd party API.

My structure is as follows, with the main path file not being edited past logging. For example, if I were to request events/test-event as the URL, it doesn't reach the expected path file.

    pages/
        api/...
        [[...path]].tsx --> file is catching all entries
        events/
            [...path]].tsx --> expected route to be caught

On testing this structure which I've used in previous projects it appears, due to the remapping of props in the middleware.ts file, the first level path file always gets the requests regardless of the URL provided.

If I disable the events/ in the matcher file for the middleware, I get a 404 for the URL but the logs in events/path fire correctly.

As I wanted to leverage the XMCloud SXA starters routing that comes out of the box, is there a suggested structure to use, or information about adapting the middleware.ts files to support this?

Many thanks for your help.

### Additional information

Using - @sitecore-jss/sitecore-jss-nextjs v^21.1.2
"next": "^13.1.6",
"react": "^18.2.0",

As set by the https://github.com/sitecorelabs/xmcloud-foundation-head project

**Comment by ambrauer** [comment, marked as answer]
Hi @Thomas-Ballard -

The latest xmcloud-foundation-head includes the [Multisite add-on](https://doc.sitecore.com/xmc/en/developers/xm-cloud/the-next-js-multisite-add-on.html) by default. Part of this is a middleware for multisite which is likely getting in your way as this rewrites the incoming path with a site segment. I see 2 options:

1. Exclude your events routes from multisite middleware (mentioned in the [docs here](https://doc.sitecore.com/xmc/en/developers/xm-cloud/the-next-js-multisite-add-on.html#:~:text=If%20you%20implemented,multisite.ts%20file.)). e.g.
```
excludeRoute: (pathname) => {
  return pathname.startsWith('/events');
},
```
2. Remove the multisite middleware/add-on entirely. If you don't plan on serving multiple sites from a single rendering host, then this is probably your best option.

**Comment by Thomas-Ballard** [comment]
Brilliant, thanks @ambrauer - this seems to be working correctly - I wanted to ensure I didn't mess up the localisation/path management incase there was a reliance on the specific path structure, so this makes sense.
